connection/src/thick.rs

```diff
@@ -319,14 +324,19 @@ impl<CP: CredentialsProvider> AttestedConnection for ThickClient<CP> {
                     .map_err(ThickClientAttestationError::from)
             })?;
 
+        let now = SystemTime::now();
+        let epoch_time = now
+            .duration_since(UNIX_EPOCH)
+            .map_err(|_| ThickClientAttestationError::Other("Time went backwards".to_owned()))?;
+        let time = DateTime::from_unix_duration(epoch_time)
+            .map_err(|_| ThickClientAttestationError::Other("Time out of range".to_owned()))?;
```

Comment:
I'll be curious what @eranrund thinks, but I wonder if this should be an `expect`.

Not sure, the error is nice to allow someone to try and handle, but I don't think it's really recoverable.

Reply:
A little bit of discussion about this above. We could get one of these errors if the system time changes and is set backwards, which should be recoverable by just retrying. But that's not really an expected event. So I would also be fine with using `expect` with a helpful message. Other options are to keep this `Other` error variant or to create one specifically for this time error.

Reply:
After thinking about it more, the error is probably a better choice, even though this really shouldn't happen.